### ISSUE TITLE
Debug targets that don't exist in launch.json

### ISSUE THREAD

**Issue by dbaeumer** [microsoft/vscode#140980]
See attached screen shot.

<img width="853" alt="capture" src="https://user-images.githubusercontent.com/1931590/150163190-3aa283eb-fa71-406b-874d-b91ff916c53d.png">

I have no idea where they are coming from and how to get rid of them. Clicking on the gear icon does nothing and no config is revealed.

**Comment by dbaeumer** [comment]
Nothing in dev console either.

**Comment by gjsjohnmurray** [comment]
These have probably been provided dynamically by the [DebugConfigurationProvider](https://code.visualstudio.com/api/references/vscode-api#DebugConfigurationProvider) of an extension, after which they get persisted with the workspace.

**Comment by dbaeumer** [comment]
Yes, the entries are there even if I start with all extensions disabled.

**Comment by weinand** [comment]
@roblourens we are not verifying on startup whether the extension which provided the dynamic configuration is still installed and enabled (see https://github.com/microsoft/vscode/issues/110318).
But according to https://github.com/microsoft/vscode/issues/110318#issuecomment-728139062 an error should be shown if an invalid configuration is run.

**Comment by gjsjohnmurray** [comment]
Maybe users should have a way to discover which extension contributed such a dynamic configuration.

**Comment by weinand** [comment]
@gjsjohnmurray yes, good idea. Showing the extension's name on hover would be one approach.

**Comment by ravindUwU** [comment]
I was able to remove the `workspaceStorage` folder as described in the answers to https://stackoverflow.com/questions/67499188. Their presence (when accidentally added) gets annoying when I try to select/cycle through launch configs by pressing a letter with the dropdown focused. **An option to delete dynamic configs _in the "Run & Debug" view_ would be ideal.** 

#197821 starts to discuss what this option could look like, e.g., a custom `<select>`. Seeing how the settings button 👇 opens `launch.json` with the cursor at the definition of the selected static (?) launch config,

![](https://github.com/user-attachments/assets/46b75f62-302a-477d-9c69-20ff01ec8a70)

...thoughts on changing the settings button to a delete button if the selected launch config is dynamic? 🤔
